src/ucp/core/ucp_ep.c

```diff
@@ -2235,3 +2240,26 @@ int ucp_ep_config_test_rndv_support(const ucp_ep_config_t *config)
     return (config->key.err_mode == UCP_ERR_HANDLING_MODE_NONE) ||
            (config->key.cm_lane  != UCP_NULL_LANE);
 }
+
+unsigned ucp_ep_check(ucp_ep_h ep)
+{
+    uint64_t lane;
+
+    if (ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) {
+        /* ep is not configured */
+        return 1; /* return 1 because EP may be configured later */
+    }
+
+    if (!ucp_ep_config(ep)->key.ep_check_map ||
+        (ep->flags & UCP_EP_FLAG_FAILED)) {
+        return 0;
+    }
+
+    ucs_for_each_bit(lane, ucp_ep_config(ep)->key.ep_check_map) {
+        ucs_assert(lane < UCP_MAX_LANES);
+        /* coverity[overrun-local] */
+        uct_ep_check(ep->uct_eps[lane], 0, NULL);
```

Comment:
check status

Reply:
nop, error handling here is not needed: in case if no resources available then we just ignore keepalive request